bin/main

```diff
@@ -1,78 +1,65 @@
-# proramme prints " #{playes1 name} your totem is O"
-# programme asks the player2 name
-# proramme prints " #{playes2 name} your totem is X"
-# programme prints the Board
-# programme asks palayer1 : "Pick an avalible number"
-# programme validates the picks if it's not avalible programme
-# prints "Sorry this number is not avalible anymore, choose an other one"
-# programme displays new updated board
-# programme asks player2 : "Pick an avalible number"
-# programme validates the picks if it's not avalible programme prints
-# "Sorry this number is not avalible anymore, choose an other one"
-# programme displays new updated board
-# ---- iterate till the end of the game----
-# 3 cases:
-#     programme displays : "player1/2 is the winner"
-#     programme displays : "Old women (According to the brazilian version)"
+# Initialize the game
 
-# Data recived from user
-#     the name of the player in a string variable player_name1/2
-#     the numbers that a player picks are stored in an integer array picks_player1/2
+player1 = Player.new('O')
+player2 = Player.new('X')
 
-puts '*********************************************'
-puts ''
-puts '******* Welcome to Tic Tac Toe game *********'
-puts ''
-puts '*********************************************'
-puts 'player 1: Please enter your name'
-player1_name = gets.chomp.capitalize
-puts ''
-puts "#{player1_name} your totem is O"
-puts ''
-puts 'player 2: Please enter your name'
-puts ''
-player2_name = gets.chomp.capitalize
-puts ''
-puts "#{player2_name} your totem is X"
-puts ''
-puts '1 | 2 | 3
----------
-4 | 5 | 6
----------
-7 | 8 | 9'
-puts ''
-puts "#{player1_name} Pick an avalible number"
-picks_player1 = []
-picks_player1 << gets.chomp.to_i
-# if the number was already choosen
-# diplay: Sorry this number is not avalible anymore, choose an other one
-# if is the number is out of the range
-# diplay: Please, choose a number avalible from 1 to 9
-# else display the board
-puts ''
-puts '1 | 2 | 3
----------
-4 | 5 | 6
----------
-7 | 8 | 9'
-puts ''
-puts "#{player2_name} Pick an avalible number"
-picks_player2 = []
-picks_player2 << gets.chomp.to_i
-# if the number was already choosen,
-#  diplay: Sorry this number is not avalible anymore, choose an other one
-# if is the number is out of the range,
-#  diplay: Please, choose a number avalible from 1 to 9
-# else display the board
-puts ''
-puts '1 | 2 | 3
----------
-4 | 5 | 6
----------
-7 | 8 | 9'
-#  ---- iterate till the end of the game----
-# puts the name of the winner or if the game is draw ('old lady')
+game = Game.new(player1, player2)
+
+puts game.welcome_message
+
+init_player = proc do |player, numb|
+  puts game.enter_player(numb)
+  player.name = gets.chomp.capitalize
+  puts game.assign_totem(player)
+end
+
+init_player.call(player1, 1)
+init_player.call(player2, 2)
+
+# playing
+
+board = Board.new
+puts game.display_grid(board.grid)
+
+validate_pick = proc do |pick|
+  until board.pick_valid?(pick) && board.not_picked?(pick)
```

Comment:
There's an issue in the validation pick method, as you can see in the image below when I choose a position that's already filled I get the message "Number already picked, Try again" at this point it's correct, but when I select a free position, the position selected by the previous player gets filled with the actual player symbol.
Tip: You can print the board every time a selection or reselection of a position is executed.
![Screen Shot 2020-06-18 at 9 38 39 PM](https://user-images.githubusercontent.com/52430951/85091176-7c87fe80-b1ac-11ea-9527-59b29e10d315.png)

Reply:
Thank for the review! issue resolved